pkg/localkube/localkube.go

```diff
@@ -71,4 +80,23 @@ func (lk LocalkubeServer) GetAPIServerInsecureURL() string {
 // Get the host's public IP address
 func (lk LocalkubeServer) GetHostIP() (net.IP, error) {
 	return utilnet.ChooseBindAddress(net.ParseIP("0.0.0.0"))
-}
\ No newline at end of file
+}
+
+func (lk LocalkubeServer) GenerateCerts() {
+
+	if util.CanReadFile(lk.GetPublicKeyCertPath()) && util.CanReadFile(lk.GetPrivateKeyCertPath()) {
+		fmt.Println("Using existing certs")
+		return
+	}
+
+	alternateIPs := []net.IP{lk.ServiceClusterIPRange.IP}
+	alternateDNS := []string{fmt.Sprintf("%s.%s", "kubernetes.default.svc", lk.DNSDomain), "kubernetes.default.svc", "kubernetes.default", "kubernetes"}
+	hostIP, err := lk.GetHostIP()
+	if err != nil {
+		fmt.Println("Failed to get host IP: ", err)
+	}
+
+	if err := utilcrypto.GenerateSelfSignedCert(hostIP.String(), lk.GetPublicKeyCertPath(), lk.GetPrivateKeyCertPath(), alternateIPs, alternateDNS); err != nil {
+		fmt.Println("Failed to create certs: ", err)
```

Comment:
The whole error handling have to be refactored and taken care of as a whole. 
The same for logging.
But updating this to return the error :)